models/moveProductDao.js

```diff
@@ -0,0 +1,29 @@
+const { database }= require('./dataSource');
+
+const requestProduct = async (productId) => {
+    const data = await database.query(`
+        SELECT 
+            p.id,
+            p.name AS productName,
+            p.description,
+            p.price,
+            i.image_url AS imageUrl,
+            c.name AS category,
+            s.name AS special,
+            g.gender,
+            t.size,
+            t.stock
+        FROM products p
+        JOIN categories c ON p.category_id = c.id
+        JOIN special s ON p.special_id = s.id
+        JOIN gender g ON p.gender_id = g.id
+        JOIN product_images i ON p.id = i.product_id
+        JOIN stock t ON p.id = t.product_id
+        WHERE p.id = ?
+    `,
+    [productId])
+
+    return data;
```

Comment:
`database.query()` 함수를 호출하면 배열 형태로 반환됩니다.
그렇기 때문에 아래와 같이 배열에 포함된 객체 하나만 꺼내서 반환해주면 함수가 반환하는 값이 더욱 명확해질 것 같습니다.

```suggestion
    const [ data ] = await database.query(`
        SELECT 
            p.id,
            p.name AS productName,
            p.description,
            p.price,
            i.image_url AS imageUrl,
            c.name AS category,
            s.name AS special,
            g.gender,
            t.size,
            t.stock
        FROM products p
        JOIN categories c ON p.category_id = c.id
        JOIN special s ON p.special_id = s.id
        JOIN gender g ON p.gender_id = g.id
        JOIN product_images i ON p.id = i.product_id
        JOIN stock t ON p.id = t.product_id
        WHERE p.id = ?
    `,
    [productId])

    return data;
```